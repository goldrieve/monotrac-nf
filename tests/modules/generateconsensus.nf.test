nextflow_process {

    name "Test Process GENERATE_CONSENSUS"
    script "main.nf"
    process "GENERATE_CONSENSUS"

    test("Should generate consensus") {

        when {
            params {
                reference = "${projectDir}/data/references/padded_targets_positive.fasta"
                orf = "$projectDir/data/references/orf.bed"

            }
            process {
                """
                input[0] = ["SVP_test", file("${projectDir}/tests/data/SVP_test_final.vcf.gz"), file("${projectDir}/tests/data/SVP_test_final.vcf.gz.csi")] 
                input[1] = params.reference
                input[2] = params.orf
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
