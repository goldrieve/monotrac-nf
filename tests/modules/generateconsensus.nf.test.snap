{
    "Should generate consensus": {
        "content": [
            {
                "0": [
                    "SVP_test_padded.fa:md5,aec96a081fb6459ab1fc42691494b517"
                ],
                "1": [
                    [
                        "SVP_test.fa:md5,582c369fb8df7e9333691d4dd15f5eca",
                        "SVP_test_padded.fa:md5,aec96a081fb6459ab1fc42691494b517"
                    ]
                ],
                "2": [
                    [
                        "SVP_test",
                        [
                            "SVP_test.fa:md5,582c369fb8df7e9333691d4dd15f5eca",
                            "SVP_test_padded.fa:md5,aec96a081fb6459ab1fc42691494b517"
                        ]
                    ]
                ],
                "fasta": [
                    [
                        "SVP_test.fa:md5,582c369fb8df7e9333691d4dd15f5eca",
                        "SVP_test_padded.fa:md5,aec96a081fb6459ab1fc42691494b517"
                    ]
                ],
                "sample_fasta": [
                    [
                        "SVP_test",
                        [
                            "SVP_test.fa:md5,582c369fb8df7e9333691d4dd15f5eca",
                            "SVP_test_padded.fa:md5,aec96a081fb6459ab1fc42691494b517"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T14:43:31.575868921"
    }
}